Android/app/src/go/doh/doh_test.go

```diff
@@ -889,3 +863,77 @@ func TestServfail(t *testing.T) {
 		t.Errorf("Wrong question: %v", servfail.Questions[0])
 	}
 }
+
+func TestQueryCanBeCancelled(t *testing.T) {
+	expectDoHTimeout := func(server testingDoHServer, msg string) {
+		doh := makeTestDoHTransport(t, server)
+		st := time.Now()
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+		doh.Query(ctx, simpleQueryBytes)
+		require.WithinRange(t, time.Now(), st.Add(1700*time.Millisecond), st.Add(2300*time.Millisecond), msg)
+	}
+
+	expectDoHTimeout(unreachableDoH, "unreachable server should timeout within deadline of ctx")
+
+	// Intentionally create a local DoH server that does not accept any requests
+	addr, err := url.Parse(localDoH.url)
+	require.NoError(t, err)
+	dohAddr, err := net.ResolveTCPAddr("tcp", addr.Host)
+	require.NoError(t, err)
+	svr, err := net.ListenTCP("tcp", dohAddr)
+	require.NoError(t, err)
+	defer svr.Close()
+
+	expectDoHTimeout(localDoH, "unresponsive server should timeout within deadline of ctx")
+}
+
+/******** Test DoH servers ********/
+type testingDoHServer struct {
```

Comment:
This is not really a "server". That confused me. Perhaps call it something like `dohConfig`?

Reply:
done